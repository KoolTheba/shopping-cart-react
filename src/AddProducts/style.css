.products-list {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
    margin-bottom: 5rem;
}

.product-image img {
    width: 150px;
    height: 150px;
}

.product-info {
    text-align: center;
}

p {
    margin-bottom: 15px;
}

.add-to-cart-button {
    cursor: pointer;
    font-weight: bold;
    width: 145px;
    height: 2.4rem;
    font-size: 1rem;
}

.add-to-cart-button:hover {
    background-color: #799F0C;
}

.cart-icon {
    margin-left: 0.5rem;
}