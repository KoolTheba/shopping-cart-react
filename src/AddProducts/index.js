import React from 'react'
import './style.css'
import data from '../data/products.json'
console.log(data)

export default class AddProducts extends React.Component {

    state = {
        products: []
    }

    componentDidMount(){
        this.setState({ products: data })
    }

    handleAddProduct = (id) => {
        console.log('ID-----', id)
    }

    render(){
        const { products } = this.state

        return(
            <>
            <h3>Cool products to add to your shopping list</h3>
            <section className='products-list'>
            {Object.entries(products).map(([id, info]) => (
                <div 
                    className="product-card"
                    id={id}
                    key={id}
                >
                    <div className="product-image">
                        <img src={info.image} alt={info.image}/>
                    </div>
                    <div className="product-info">
                        <p>{info.name}</p>
                        <p>â‚¬{info.price}</p>
                        <button
                            onClick={() => this.handleAddProduct(id)}
                        >Add to Cart</button>
                    </div>
                </div>
            ))}
            </section>
            </>
        )
    }
}