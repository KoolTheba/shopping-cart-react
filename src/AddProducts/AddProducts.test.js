import React from 'react'
import AddProducts from './'
import { shallow } from 'enzyme'

const props = {
    products: {
        "PLANT14TGG": {
            "name": "Zamioculcas",
            "price": 90.00,
            "stock": true,
            "stockQ": 15,
            "image": "https://image.shutterstock.com/image-photo/pot-zamioculcas-home-plant-on-600w-1399949501.jpg"
        },
    },
    addProducts: jest.fn()
}

describe('AddProducts component', () => {
    it('renders AddProducts component correctly', () => {
        const wrapper = shallow(<AddProducts {...props}/>)
        expect(wrapper.find('.products-list')).toHaveLength(1)
        expect(wrapper.find('.product-card')).toHaveLength(1)
        expect(wrapper).toMatchSnapshot()
    });

    it('when clicking add to cart button, addProducts() is called', () => {
        const wrapper = shallow(<AddProducts {...props}/>)
        const addToCartButton = wrapper.find('.add-to-cart-button')
        addToCartButton.simulate('click')
        expect(props.addProducts).toHaveBeenCalledTimes(1)
    });
});