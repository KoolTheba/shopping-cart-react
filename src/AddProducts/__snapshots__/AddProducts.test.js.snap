// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProducts component renders AddProducts component correctly 1`] = `ShallowWrapper {}`;
